fastlane/example.env

```diff
@@ -0,0 +1,8 @@
+# You don't need to fill in these values for Fastlane to run.
+#
+# However, if a lane requires some of these environment values and they are not set here, it will fail.
```

Comment:
What lane would require these?

Reply:
The `APP_STORE_*` env vars are required by any lanes interacting with App Store Connect, such as the one uploading to TestFlight.

The `MATCH_*` ones by the code signing sync lane.

I've added calls to `require_env_vars!(...)` in each lane to ideally document in code the hidden env var dependencies as well as fail with a useful message.